### PR TITLE
Make deprecation warning messages more useful

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

The current warning messages may be less kind because:

- Users do not know how to fix such warnings.
- Users do not have time to read the documents.

So, this change aims to try reducing users' potential complaints.

> Link related issues, e.g. "Fix #<ISSUE_ID>", "Related to #<ISSUE_ID>", or "None."

None.

> Check the following items.

- [x] Add a new [changelog](https://github.com/sider/runners/blob/master/CHANGELOG.md) entry if this change is notable.

### CODE REVIEW

File: lib/runners/processor.rb [hunk 1]
```diff
@@ -177,18 +177,35 @@ def add_warning_if_deprecated_version(minimum:, file: nil, deadline: nil)
...
-          DEPRECATION WARNING!!!
-          The following options in your `#{config.path_name}` are deprecated and will be removed.
-          See #{analyzer_doc} for details.
-          #{deprecated_keys.join("\n")}
-        MSG
-      end
+      file = config.path_name
+      add_warning <<~MSG, file: file
+        DEPRECATION WARNING!!!
+        The `#{config_field_path(name)}` option is deprecated in your `#{file}`. Fix it as follows:
```
Comment (yykamei) [inline lib/runners/processor.rb:187]: How about this? I think the option is deprecated for Sider in whatever file.

```suggestion
        The `#{config_field_path(name)}` option is deprecated. Fix your `#{file}` as follows:
```

File: lib/runners/processor.rb [hunk 2]
```diff
@@ -177,18 +177,35 @@ def add_warning_if_deprecated_version(minimum:, file: nil, deadline: nil)
...
+      MSG
+    end
+
+    def add_warning_for_deprecated_option(name, to:)
+      return unless config_linter[name]
+
+      file = config.path_name
+      add_warning <<~MSG, file: file
+        DEPRECATION WARNING!!!
+        The `#{config_field_path(name)}` option is deprecated in your `#{file}`. Use the `#{config_field_path(to)}` option instead.
```
Comment (yykamei) [inline lib/runners/processor.rb:206]:
```suggestion
        The `#{config_field_path(name)}` option is deprecated. Use the `#{config_field_path(to)}` option instead.
```

File: lib/runners/processor.rb (hunk 1 above)
Comment (ybiquitous) [reply lib/runners/processor.rb]: Nice idea! 👍

File: lib/runners/processor.rb (hunk 2 above)
Comment (ybiquitous) [reply lib/runners/processor.rb:206]: I doubt some users may not know what should be fixed if there is no `sider.yml` word in the warning message. 🤔

Comment (ybiquitous) [reply lib/runners/processor.rb:206]: For a better idea, we may need to improve the Web UI. We have not used the `file` value in a `warning` hash. 🤔

Comment (yykamei) [reply lib/runners/processor.rb:206]: If you want to add `#{file}` in the sentence, I guess the below is sufficient.

```suggestion
        The `#{config_field_path(name)}` option is deprecated in your `#{file}`. Use the `#{config_field_path(to)}` option instead in your `#{file}`.
```

File: test/processor_test.rb [hunk 3]
```diff
@@ -244,20 +244,24 @@ def test_add_warning_if_deprecated_options
...
       YAML
 
-      processor.add_warning_if_deprecated_options([:quiet, :options])
-      processor.add_warning_if_deprecated_options([:global])
+      processor.add_warning_if_deprecated_options
 
       expected_message = <<~MSG.strip
         DEPRECATION WARNING!!!
-        The following options in your `sider.yml` are deprecated and will be removed.
+        The `linter.eslint.options` option is deprecated in your `sider.yml`. Fix it as follows:
```
Comment (ybiquitous) [inline test/processor_test.rb:255]:
```suggestion
        The `linter.eslint.options` option is deprecated. Fix your `sider.yml` as follows:
```

File: lib/runners/processor.rb (hunk 2 above)
Comment (ybiquitous) [reply lib/runners/processor.rb:206]: Do you mean to repeat `` in your `#{file}` ``?

Comment (yykamei) [reply lib/runners/processor.rb:206]: No, sorry, this is correct.
```suggestion
        The `#{config_field_path(name)}` option is deprecated. Use the `#{config_field_path(to)}` option instead  in your `#{file}`.
```

Comment (ybiquitous) [reply lib/runners/processor.rb:206]: Ah, sounds good. 👍

File: test/processor_test.rb [hunk 4]
```diff
@@ -271,6 +275,33 @@ def test_add_warning_if_deprecated_options
...
+        linter:
+          eslint:
+            dir: src
+      YAML
+
+      processor.add_warning_for_deprecated_option(:dir, to: :target)
+
+      expected_message = <<~MSG.strip
+        DEPRECATION WARNING!!!
+        The `linter.eslint.dir` option is deprecated in your `sider.yml`. Use the `linter.eslint.target` option instead.
```
Comment (ybiquitous) [inline test/processor_test.rb:290]:
```suggestion
        The `linter.eslint.dir` option is deprecated. Use the `linter.eslint.target` option instead in your `sider.yml`.
```

File: test/smokes/eslint/expectations.rb [hunk 5]
```diff
@@ -254,7 +249,13 @@
     }
   ],
   analyzer: { name: "ESLint", version: "5.16.0" },
-  warnings: [{ message: "The `dir` option is deprecated. Use the `target` option instead.", file: "sideci.yml" }]
+  warnings: [
+    { message: <<~MSG.strip, file: "sideci.yml" }
+      DEPRECATION WARNING!!!
+      The `linter.eslint.dir` option is deprecated in your `sideci.yml`. Use the `linter.eslint.target` option instead.
```
Comment (ybiquitous) [inline test/smokes/eslint/expectations.rb:255]:
```suggestion
      The `linter.eslint.dir` option is deprecated. Use the `linter.eslint.target` option instead in your `sideci.yml`.
```

File: test/smokes/haml_lint/expectations.rb [hunk 6]
```diff
@@ -35,18 +35,17 @@
...
   warnings: [
+    {
+      message: /The `linter.haml_lint.options` option is deprecated/,
+      file: "sideci.yml"
+    },
     {
       message: <<~MSG.strip,
         DEPRECATION WARNING!!!
-        The following options in your `sideci.yml` are deprecated and will be removed.
+        The `linter.haml_lint.file` option is deprecated in your `sideci.yml`. Use the `linter.haml_lint.target` option instead.
```
Comment (ybiquitous) [inline test/smokes/haml_lint/expectations.rb:45]:
```suggestion
        The `linter.haml_lint.file` option is deprecated. Use the `linter.haml_lint.target` option instead in your `sideci.yml`.
```

File: test/smokes/misspell/expectations.rb [hunk 7]
```diff
@@ -142,9 +137,8 @@
   analyzer: { name: "Misspell", version: default_version },
   warnings: [{ message: <<~MSG.strip, file: "sideci.yml" }]
     DEPRECATION WARNING!!!
-    The following options in your `sideci.yml` are deprecated and will be removed.
+    The `linter.misspell.targets` option is deprecated in your `sideci.yml`. Use the `linter.misspell.target` option instead.
```
Comment (ybiquitous) [inline test/smokes/misspell/expectations.rb:140]:
```suggestion
    The `linter.misspell.targets` option is deprecated. Use the `linter.misspell.target` option instead in your `sideci.yml`.
```

**Pull request merged**